### PR TITLE
Fix flakey tests failing in CI

### PR DESCRIPTION
**Pull request by astubbs**

### CODE REVIEW

File: .idea/runConfigurations/All.xml [hunk 1]
```diff
@@ -1,21 +1,20 @@
-<component name="ProjectRunConfigurationManager">
```
Comment (astubbs) [inline .idea/runConfigurations/All.xml:1]: why is this changed? CRLF issues?

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/offsets/OffsetSimultaneousEncoder.java [hunk 2]
```diff
@@ -43,12 +43,9 @@
...
     /**
-     * The next expected offset to be returned by the broker
-     */
-    private final long nextExpectedOffset;
-
-    /**
-     * The difference between the base offset (the offset to be committed) and the highest seen offset
+     * The difference between the base offset (the offset to be committed) and the highest seen offset.
+     * <p>
+     * {@link BitSet} only support {@link Integer#MAX_VALUE) bits
```
Comment (astubbs) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/offsets/OffsetSimultaneousEncoder.java:48]:
```suggestion
     * {@link BitSet} only supports {@link Integer#MAX_VALUE) bits
```

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/state/WorkManager.java [hunk 3]
```diff
@@ -473,13 +479,21 @@ public boolean hasCommittableOffsets() {
             var workToRemove = new LinkedList<WorkContainer<K, V>>();
             var incompleteOffsets = new LinkedHashSet<Long>();
             long lowWaterMark = -1;
+            var highestSucceeded = partitionState.getOffsetHighestSucceeded();
             // can't commit this offset or beyond, as this is the latest offset that is incomplete
             // i.e. only commit offsets that come before the current one, and stop looking for more
-            boolean iteratedBeyondLowWaterMarkBeingLowestCommittableOffset = false;
+            boolean iteratedBeyondLowWaterMarkBeingLowestCommittableOffset = false; // todo terrible name
```
Comment (astubbs) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/state/WorkManager.java:485]: rename

File: parallel-consumer-core/src/test-integration/java/io/confluent/parallelconsumer/integrationTests/MultiInstanceRebalanceTest.java [hunk 4]
```diff
@@ -132,7 +132,9 @@ private void runTest(int maxPoll, CommitMode commitMode, ProcessingOrder order)
                     .alias(failureMessage)
                     .pollInterval(1, SECONDS)
                     .untilAsserted(() -> {
-                        log.trace("Processed-count: {}", getAllConsumedKeys(pc1, pc2).size());
+                        log.info("Processed-count: {}", getAllConsumedKeys(pc1, pc2).size());
+                        // TODO ok to reenable?
```
Comment (astubbs) [inline parallel-consumer-core/src/test-integration/java/io/confluent/parallelconsumer/integrationTests/MultiInstanceRebalanceTest.java:136]: seems ok in CI

File: parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/offsets/OffsetEncodingBackPressureTest.java [hunk 5]
```diff
@@ -178,7 +193,10 @@ void backPressureShouldPreventTooManyMessagesBeingQueuedForProcessing() throws O
                             String meta = partitionCommit.metadata();
                             HighestOffsetAndIncompletes incompletes = OffsetMapCodecManager
                                     .deserialiseIncompleteOffsetMapFromBase64(0L, meta);
-                            assertThat(incompletes.getIncompleteOffsets()).containsOnly(0L);
+                            // todo dep issue with truth? uncomment
+//                            Truth.assertWithMessage("The only incomplete record now is offset zero, which we are blocked on")
```
Comment (astubbs) [inline parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/offsets/OffsetEncodingBackPressureTest.java:197]: fix this

File: parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/offsets/OffsetEncodingBackPressureTest.java [hunk 6]
```diff
@@ -159,17 +160,31 @@ void backPressureShouldPreventTooManyMessagesBeingQueuedForProcessing() throws O
             // assert partition now blocked from threshold
             int expectedMsgsProcessedBeforePartitionBlocks = numRecords + numRecords / 4;
             {
+
+                // todo WIP fix for race condition to force the test be consistent, regardless of race - change to event based / awaitility
```
Comment (astubbs) [inline parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/offsets/OffsetEncodingBackPressureTest.java:164]: improve this

File: parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/offsets/OffsetEncodingTests.java [hunk 7]
```diff
@@ -150,11 +148,16 @@ void ensureEncodingGracefullyWorksWhenOffsetsAreVeryLargeAndNotSequential(Offset
...
         records.add(new ConsumerRecord<>(INPUT_TOPIC, 0, 40_000 + Short.MAX_VALUE + avoidOffByOne, "akey", "avalue")); // runlength higher than Short.MAX_VALUE
+        int highest = 40_000 + Short.MAX_VALUE + avoidOffByOne + 1;
+        records.add(new ConsumerRecord<>(INPUT_TOPIC, 0, highest, "akey", "avalue")); // will complete to force whole encoding
+
+
+        var incompleteRecords = new ArrayList<>(records);
+        incompleteRecords.remove(incompleteRecords.stream().filter(x -> x.offset() == 0).findFirst().get());
+        incompleteRecords.remove(incompleteRecords.stream().filter(x -> x.offset() == 69).findFirst().get());
+        incompleteRecords.remove(incompleteRecords.stream().filter(x -> x.offset() == 25_000).findFirst().get());
+        incompleteRecords.remove(incompleteRecords.stream().filter(x -> x.offset() == highest).findFirst().get());
```
Comment (astubbs) [inline parallel-consumer-core/src/test/java/io/confluent/parallelconsumer/offsets/OffsetEncodingTests.java:159]: extract record numbers - make DRY

File: .editorconfig [hunk 8]
```diff
@@ -0,0 +1,892 @@
+[*]
```
Comment (astubbs) [inline .editorconfig:1]: delete

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/state/WorkManager.java [hunk 9]
```diff
@@ -368,6 +370,10 @@ public void onSuccess(WorkContainer<K, V> wc) {
         ConsumerRecord<K, V> cr = wc.getCr();
         log.trace("Work success ({}), removing from processing shard queue", wc);
         wc.succeed();
+
+        // update as we go
+        pm.getState(wc.getTopicPartition()).updateHighestSucceededOffsetSoFar(wc);
```
Comment (astubbs) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/state/WorkManager.java:375]: leaky, cohesion, change to calling `PartitionState#onSuccess`

File: .idea/runConfigurations/All.xml (hunk 1 above)
Comment (astubbs) [reply .idea/runConfigurations/All.xml:1]: yes. that and removing the license check. removed that to another commit, but the whitespace fix remains and its tricky to remove so will leave it in.

**Review by rkolesnev — APPROVED** [review]
